main.py

```diff
@@ -41,99 +40,175 @@ def check(a, b):
     return new
 
 
-def clean_board():
-    """Returns a new, blank board"""
-    # noinspection PyUnusedLocal
-    return [[Cell() for a in range(config.Height + (2 * config.Cushion))] for b in
-            range(config.Width + 2 * config.Cushion)]
-
-
 def draw(state, a, b, colour):
     """Draws a type of cell (Type) at the desired cell (a,b)"""
-    x = (a - config.Cushion) * config.Size + config.Edge / 2
-    y = (b - config.Cushion) * config.Size + config.Edge / 2
-    s = config.Size - config.Edge
+    x = (a - Board.Cushion) * Board.Size + Board.Edge / 2
+    y = (b - Board.Cushion) * Board.Size + Board.Edge / 2
+    s = Board.Size - Board.Edge
     pygame.draw.rect(Screen, (255, 255, 255), (x, y, s, s))
     if state == config.Square:
         pygame.draw.rect(Screen, colour, (x, y, s, s))
 
 
-# noinspection PyShadowingNames
-def check_user_input(board, paused):
+# noinspection PyUnresolvedReferences
+def check_user_input(board, game_state):
     """Checks for user input and acts accordingly"""
-    one_turn = False
     for event in pygame.event.get():
         x, y = pygame.mouse.get_pos()
-        a = maths.floor(x / config.Size) + config.Cushion
-        b = maths.floor(y / config.Size) + config.Cushion
+        a = x // board.Size + board.Cushion
+        b = y // board.Size + board.Cushion
         if pygame.key.get_pressed()[pygame.K_SPACE]:
-            if paused:
-                paused = False
-            else:
-                paused = True
+            game_state.Paused = not game_state.Paused
         if event.type == pygame.QUIT or pygame.key.get_pressed()[pygame.K_ESCAPE]:
             pygame.quit()
             import sys
             sys.exit(0)
         for key in range(pygame.K_1, pygame.K_8):
             if pygame.key.get_pressed()[key]:
                 board = preset.place(board, int(pygame.key.name(key)), a, b)
+                board = update_board(board)
+                draw_board(board)
+        if pygame.key.get_pressed()[pygame.K_f]:
+            game_state.GPSLimit = not game_state.GPSLimit
+            #draw_gps_slider(((maths.log(game_state.GPS, game_state.TopGPS) + 1) / -3) * (
+            #Widgets.EndOfSlider - Widgets.StartOfSlider) +
+            #                Widgets.EndOfSlider, game_state.GPSLimit)
+            bottom_gps_log=maths.log(game_state.BottomGPS,game_state.TopGPS)
+            draw_gps_slider(Widgets.EndOfSlider-((maths.log(game_state.GPS,game_state.TopGPS)-bottom_gps_log)*
+                                                 (Widgets.EndOfSlider-Widgets.StartOfSlider))/
+                            (1-bottom_gps_log),game_state.GPSLimit)
         if pygame.key.get_pressed()[pygame.K_RIGHT]:
-            one_turn = True
+            game_state.OneTurn = True
+        else:
+            game_state.OneTurn = False
         if pygame.key.get_pressed()[pygame.K_RETURN]:
-            board = clean_board()
+            board = config.Board()
+            board = update_board(board)
```

Comment:
You create a `board` and then immediately overwrite it with `update_board`?

Reply:
done